managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1711,7 +1711,7 @@ public void asyncMarkDelete(final Position position, Map<String, Long> propertie
                 long ledgerEntries = ledger.getLedgerInfo(markDeletePosition.getLedgerId()).get().getEntries();
                 Long nextValidLedger = ledger.getNextValidLedger(ledger.getLastConfirmedEntry().getLedgerId());
                 shouldCursorMoveForward = (markDeletePosition.getEntryId() + 1 >= ledgerEntries)
-                        && (newPosition.getLedgerId() == nextValidLedger);
+                        && (nextValidLedger != null && newPosition.getLedgerId() == nextValidLedger);
```

Comment:
I think we can check ``nextValidLedger != null`` first and then

```java
(markDeletePosition.getEntryId() + 1 >= ledgerEntries)
                        && (newPosition.getLedgerId() == nextValidLedger);
```

Reply:
Ah, You're right, I have changed it back now.